test/integration/podgc/podgc_test.go

```diff
@@ -160,7 +160,7 @@ func TestTerminatingOnOutOfServiceNode(t *testing.T) {
 			wantPhase:                     v1.PodFailed,
 		},
 		"pod has phase when PodDisruptionConditions disabled": {
-			enablePodDisruptionConditions: true,
+			enablePodDisruptionConditions: false,
```

Comment:
This is actually a bug in the test that could cause flakiness, as with `enablePodDisruptionConditions: true` the phase is transitioned from Pending to Failed. The only reason the test was passing is that we are using `PollImmediate` which makes the check before PodGC changes the phase. Since the PodGC check period is 500ms (https://github.com/kubernetes/kubernetes/blob/3e2840400880d7e698fdb390a9cb8eef08464ce9/test/integration/podgc/podgc_test.go#L293) it is very unlikely the phase would change actually, causing the test flaky, it is just essentially ignored. 

I think in order to make the test useful we should also change the `PollImmediate` to `Poll` so that we pass a single gc check interval. We could also consider lowering the poll interval to 600ms.

We could also consider cherry-picking the fix for the test, but given failure is very unlikely, I feel there is no such need. @alculquicondor wdyt?

Reply:
Yeah, cherry-picking is worth.

Reply:
yes please